libkbfs/folder_branch_ops.go

```diff
@@ -1206,6 +1206,15 @@ func (fbo *folderBranchOps) initMDLocked(
 		return err
 	}
 
+	fbo.headLock.Lock(lState)
+	defer fbo.headLock.Unlock(lState)
+	// Some other thread got here first, so give up and let it go
+	// before we push anything to the servers.
```

Comment:
`head` is protected by `headLock`.  I guess technically since we have the writer lock there would have to be some other bug involved for something else to set the head in the meantime, but seems like bad practice not to hold the lock here.